client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx

```diff
@@ -110,29 +111,43 @@ export default function DomainForwardingCard( { domain }: { domain: ResponseDoma
 			const url = new URL( forwarding.targetPath, origin );
 			if ( url.hostname !== '_invalid_.domain' ) {
 				setTargetUrl( url.hostname + url.pathname + url.search + url.hash );
-				setProtocol( forwarding.isSecure ? 'https' : 'http' );
 				setIsPermanent( forwarding.isPermanent );
 				setForwardPaths( forwarding.forwardPaths );
+				setSubdomain( forwarding.subdomain );
+				setSourceType( forwarding.subdomain !== '' ? 'subdomain' : 'domain' );
 			}
 		} catch ( e ) {
 			// ignore
 		}
-	}, [ isLoading, forwarding, setTargetUrl, setProtocol ] );
+	}, [ isLoading, forwarding, setTargetUrl ] );
 
-	const handleChange = ( event: React.ChangeEvent< HTMLInputElement > ) => {
-		setTargetUrl( withoutHttp( event.target.value ) );
+	const handleSubdomainChange = ( event: React.ChangeEvent< HTMLInputElement > ) => {
+		const subdomain = event.target.value;
+		setSubdomain( withoutHttp( subdomain ) );
 
-		if (
-			event.target.value.length > 0 &&
-			! CAPTURE_URL_RGX_SOFT.test( protocol + '://' + event.target.value )
-		) {
+		const CAPTURE_SUBDOMAIN_RGX = /^(?!-)[a-zA-Z0-9-]{0,63}(?<!-)$/i;
```

Comment:
I updated the regex so that hyphens are not allowed at the start and end in the subdomain according to RFC 1034, section 3.5 (and if you saved that, it also won't work)

> They must
start with a letter, end with a letter or digit, and have as interior
characters only letters, digits, and hyphen.